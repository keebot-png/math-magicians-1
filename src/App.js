import React from 'react';
import { Routes, Route } from 'react-router-dom';
import Calculator from './components/calculatorPage/calculator';
import Quotes from './components/quotePage/quote';
import HomePage from './components/homePage/home';
import Navbar from './components/navbarHeading/navbar';

/* eslint-disable */
class App extends React.PureComponent {
  render() {
    return (
      <div>
    <Navbar />
    <Routes>
      <Route path="/" element={<HomePage />} />
      <Route path="/calculator" element={<Calculator />} />
      <Route path="/quote" element={<Quotes />} />
    </Routes>
  </div>
    );
  }
}

export default App;
