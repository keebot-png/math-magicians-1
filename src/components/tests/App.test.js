import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../../App';

test('Jest snapshot to test components', () => {
  const component = renderer.create(
    <Router>
      <App />
    </Router>,
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});