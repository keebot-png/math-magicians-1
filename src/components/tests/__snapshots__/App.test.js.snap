// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest snapshot to test components 1`] = `
<div>
  <nav>
    <div
      className="headline"
    >
      Math Magicians
    </div>
    <ul
      className="navList"
    >
      <li
        className="navList-item"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        |
      </li>
      <li
        className="navList-item"
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        |
      </li>
      <li
        className="navList-item"
      >
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="home-container"
  >
    <h1
      className="title"
    >
      Welcome to our page!
    </h1>
    <br />
    <p
      className="home-text"
    >
      Here is a calculator app that has successfully been built using the React framework. It has all the necessary and basic components that are suited for essential math functions.
    </p>
    <br />
    <p
      className="home-text"
    >
      The home page provides an explaination of the webapp features, which includes: a functional calculator and a random quote generator that changes every time the navbar link is clicked.
    </p>
  </div>
</div>
`;
